AppleMusic/Source/View/SearchView/UIKitSearch/UIKitSearchView.swift

```diff
@@ -0,0 +1,85 @@
+        return collection
+    }()
+    
+    override func viewDidLoad() {
+        view.backgroundColor = .white
+        view.addSubview(searchCollection)
+        
+        searchCollection.centerYAnchor.constraint(equalTo: view.centerYAnchor).isActive = true
+        searchCollection.centerXAnchor.constraint(equalTo: view.centerXAnchor).isActive = true
+        searchCollection.bottomAnchor.constraint(equalTo: view.safeAreaLayoutGuide.bottomAnchor).isActive = true
+        
+        searchCollection.widthAnchor.constraint(equalTo: view.safeAreaLayoutGuide.widthAnchor).isActive = true
+        searchCollection.heightAnchor.constraint(equalTo: view.safeAreaLayoutGuide.heightAnchor).isActive = true
+    }
+    
+    private func createCollectionLayout() -> UICollectionViewLayout {
+        return UICollectionViewCompositionalLayout { [self](sectionIndex, layoutEnvironment) -> NSCollectionLayoutSection? in
+            return createCollectionLayoutSection(count: collectionDara.count)
+            }
+    }
+    
+    
+    func createCollectionLayoutSection(count: Int) -> NSCollectionLayoutSection {
+            let spacing: CGFloat = 5
+            let bottomSpacing: CGFloat = 10
+
+            let itemSize = NSCollectionLayoutSize(widthDimension: .fractionalWidth(1),
+                                                  heightDimension: .fractionalHeight(1))
+            let item = NSCollectionLayoutItem(layoutSize: itemSize)
+            item.contentInsets = .init(top: 0, leading: spacing, bottom: bottomSpacing, trailing: spacing)
+
+        let groupSize = NSCollectionLayoutSize(widthDimension: .fractionalWidth(0.5),
+                                               heightDimension: .fractionalHeight(0.9))
+            let group = NSCollectionLayoutGroup.vertical(layoutSize: groupSize, subitem: item, count: 4)
+
+            let section = NSCollectionLayoutSection(group: group)
+        section.orthogonalScrollingBehavior = .groupPaging
+
+            return section
+        }
+    
+}
+
+extension SearchViewController: UICollectionViewDelegate {
+    func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
+        collectionView.deselectItem(at: indexPath, animated: true)
+    }
+}
+
+extension SearchViewController: UICollectionViewDataSource {
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        return collectionDara.count
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        let image = UIImage(named: collectionDara[indexPath.row])
+        guard let cell = collectionView.dequeueReusableCell(withReuseIdentifier: CategoriesCell.identifier , for: indexPath) as? CategoriesCell else { return  UICollectionViewCell() }
+        cell.imageView.image = image
+        return cell
```

Comment:
Объявление и инициализацию изображения лучше делать после того, как убедишься, что `cell` существует.
```suggestion
        guard let cell = collectionView.dequeueReusableCell(withReuseIdentifier: CategoriesCell.identifier , for: indexPath) as? CategoriesCell else { return  UICollectionViewCell() }
        let image = UIImage(named: collectionDara[indexPath.row])
        cell.imageView.image = image
        return cell
```